bgpd/bgp_route.c

```diff
@@ -2237,11 +2240,21 @@ int subgroup_process_announce_selected(struct update_subgroup *subgrp,
 	/* It's initialized in bgp_announce_check() */
 
 	/* Announcement to the subgroup.  If the route is filtered withdraw it.
+	 * If BGP_NODE_FIB_INSTALL_PENDING is set and data plane install status
+	 * is pending (BGP_NODE_FIB_INSTALL_PENDING), do not advertise the
+	 * route
 	 */
+	if (BGP_SUPPRESS_FIB_ENABLED(bgp) &&
+	    CHECK_FLAG(rn->flags, BGP_NODE_FIB_INSTALL_PENDING) &&
+			(!bgp_option_check(BGP_OPT_NO_FIB)))
+		advertise = false;
+
 	if (selected) {
-		if (subgroup_announce_check(rn, selected, subgrp, p, &attr))
-			bgp_adj_out_set_subgroup(rn, subgrp, &attr, selected);
-		else
+		if (subgroup_announce_check(rn, selected, subgrp, p, &attr)) {
+			if (advertise)
+				bgp_adj_out_set_subgroup(rn, subgrp, &attr,
+							 selected);
+		} else
 			bgp_adj_out_unset_subgroup(rn, subgrp, 1,
 						   addpath_tx_id);
```

Comment:
Withdraw is not sent, nothing is done for the route

Reply:
At the point where this check is performed, there is no chance the route has already been advertised? I've traced it through a couple of times, and it looks like that is right, but just want to make certain. If not, then we don't need to withdraw, as it would not have been advertised.

Reply:
> At the point where this check is performed, there is no chance the route has already been advertised? I've traced it through a couple of times, and it looks like that is right, but just want to make certain. If not, then we don't need to withdraw, as it would not have been advertised.

If the route was already advertised before (suppress fib is enabled and route installed in FIB) and configuration changes (suppress fib is disabled), then it will require clearing the peer for new configuration to take effect on the previously advertised routes.  Without peer reset, the configuration change will have effect on routes that are newly received received.

Reply:
> Withdraw is not sent, nothing is done for the route

@kssoman can you pleaes confirm at this point if route was not already advertised and remains pending then do we need to check if withdraw needs to be sent?

Reply:
If the route is selected but not advertised (pending flag set) to peers, then withdraw need not be sent